src/Common/LatencyBuckets.cpp

```diff
@@ -0,0 +1,106 @@
+#include <atomic>
+#include <Common/logger_useful.h>
+#include <Common/LatencyBuckets.h>
+
+#define APPLY_FOR_EVENTS(M) \
+    M(S3FirstByteReadAttempt1Microseconds, "Time of first byte read from S3 storage (attempt 1).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3FirstByteWriteAttempt1Microseconds, "Time of first byte write to S3 storage (attempt 1).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3FirstByteReadAttempt2Microseconds, "Time of first byte read from S3 storage (attempt 2).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3FirstByteWriteAttempt2Microseconds, "Time of first byte write to S3 storage (attempt 2).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3FirstByteReadAttemptNMicroseconds, "Time of first byte read from S3 storage (attempt N).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3FirstByteWriteAttemptNMicroseconds, "Time of first byte write to S3 storage (attempt N).", 100, 1000, 10000, 100000, 300000, 500000, 1000000, 2000000, 5000000, 10000000, 15000000, 20000000, 25000000, 30000000, 35000000) \
+    M(S3ReadConnectMicroseconds, "Time to connect for GET and HEAD requests to S3 storage.", 100, 1000, 10000, 100000, 200000, 300000, 500000, 1000000, 1500000) \
```

Comment:
When we connect to the server, it does not know the purpose yet. Why do we distinguish read and write connect time?

Reply:
When we write the request to the server it also does not know the request yet.

The difference between read and write requests could be significant when we reading the response from the server.

Reply:
Yeah, you're right, it's an artifact of differentiating for other metrics, made one metric for `connect`, thanks